src/components/issues/SortIssues.js

```diff
@@ -1,16 +1,208 @@
+
+type ConnectedProps = {
+  setSortStateInState: () => void,
+  setSortingInState: () => void
+};
+
+type OwnProps = {};
+
+type State = {
+  isOpen: boolean
+};
+
+class SortIssues extends Component<ConnectedProps & OwnProps, State> {
+  state = {
+    isOpen: false
+  };
+
+  setFetchByState = issuesState => {
+    this.props.setSortStateInState(issuesState);
+  };
+
+  setFetchBySort = sorting => {
+    this.props.setSortingInState(sorting);
+    this.setState({
+      isOpen: false
+    });
+  };
+
+  toggleSort = () => {
+    this.setState({
+      isOpen: !this.state.isOpen
+    });
+  };
+
+  handleClickOutSide = () => {
+    this.setState({
+      isOpen: false
+    });
+  };
+
+  render() {
+    return (
+      <SortContainer>
+        <OpenClosedContainer>
+          <Open onClick={() => this.setFetchByState('open')}>
+            <OpenIcon />
+            <OpenTitle>Open</OpenTitle>
+          </Open>
+          <Closed onClick={() => this.setFetchByState('closed')}>
+            <ClosedIcon />
+            <ClosedTitle>Closed</ClosedTitle>
+          </Closed>
+        </OpenClosedContainer>
+        <OutsideClickHandler onOutsideClick={this.handleClickOutSide}>
+          <SortSelect onClick={this.toggleSort}>
+            <SortText>Sort</SortText>
+            <SortIcon />
+          </SortSelect>
+          <DropDownContainer isOpen={this.state.isOpen}>
+            <TitleContainer>
+              <Title>Sort by</Title>
+            </TitleContainer>
+            <OptionContainer onClick={() => this.setFetchBySort('created')}>
+              <OptionTitle>Newest</OptionTitle>
+            </OptionContainer>
+            <OptionContainer onClick={() => this.setFetchBySort('created-asc')}>
+              <OptionTitle>Oldest</OptionTitle>
+            </OptionContainer>
+            <OptionContainer onClick={() => this.setFetchBySort('comments')}>
+              <OptionTitle>Most commented</OptionTitle>
+            </OptionContainer>
+            <OptionContainer
+              onClick={() => this.setFetchBySort('comments-asc')}>
+              <OptionTitle>Least comments</OptionTitle>
+            </OptionContainer>
+            <OptionContainer onClick={() => this.setFetchBySort('updated')}>
+              <OptionTitle>Recently updated</OptionTitle>
+            </OptionContainer>
+          </DropDownContainer>
+        </OutsideClickHandler>
+      </SortContainer>
+    );
+  }
+}
 
 const SortContainer = styled.div`
   background-color: #f7f8fb;
   height: 50px;
   border: 1px solid #d1d5da;
   border-radius-top-right: 3px;
   border-radius-top-left: 3px;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  position: relative;
 `;
 
-const SortIssues = () => {
-  return <SortContainer />;
-};
+const OpenClosedContainer = styled.div`
+  display: flex;
+`;
+
+const Open = styled.div`
+  display: flex;
+  align-items: center;
+  margin-left: 5px;
+  cursor: pointer;
+`;
+
+const OpenTitle = styled.h3`
```

Comment:
if you are reusing this for other components with empty definitions consider a different more generic name